go/vt/vttablet/tabletserver/txthrottler/tx_throttler.go

```diff
@@ -50,8 +50,13 @@ var (
 )
 
 func resetTxThrottlerFactories() {
-	healthCheckFactory = func(topoServer *topo.Server, cell string, cellsToWatch []string) discovery.HealthCheck {
-		return discovery.NewHealthCheck(context.Background(), discovery.DefaultHealthCheckRetryDelay, discovery.DefaultHealthCheckTimeout, topoServer, cell, strings.Join(cellsToWatch, ","))
+	healthCheckFactory = func(topoServer *topo.Server, cell, keyspace, shard string, cellsToWatch []string) (discovery.HealthCheck, error) {
+		// discovery.NewFilterByShard expects a single-shard filter to be in "keyspace|shard" format.
+		filter, err := discovery.NewFilterByShard([]string{keyspace + "|" + shard})
+		if err != nil {
+			return nil, err
+		}
+		return discovery.NewHealthCheck(context.Background(), discovery.DefaultHealthCheckRetryDelay, discovery.DefaultHealthCheckTimeout, topoServer, cell, strings.Join(cellsToWatch, ","), filter), nil
```

Comment:
I'm curious how this work is cancelled and cleaned up w/o a cancelable context? It's not new here in this PR, I know.

Reply:
@mattlord I don't have context (pun intended) on that. @ejortegau do you?

Reply:
@mattlord it appears that context [is passed to the undelying topo watcher only](https://github.com/vitessio/vitess/blob/149bf92fe90a1bc1592d8cf444c567ffad083864/go/vt/discovery/healthcheck.go#L383). [Context on this change is here](https://github.com/vitessio/vitess/pull/10542)

Reply:
@mattlord reworked the way `context.Context` is passed [here](https://github.com/vitessio/vitess/pull/16170/commits/1f7744ee1540cdb6aebf604f6a2df4ec0914acba), possibly just for clarity. Now the state struct that lives between `.Open()` and `.Close()`-only holds a context/cancel func